### PR TITLE
Calculate the nonce

### PR DESCRIPTION
**Pull request by posgnu**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -3,12 +3,25 @@ import * as sleep from "sleep";
...
     balance: BigNumber;
 }
 
+async function calculateNonce(sdk: SDK, payer: string): Promise<U256> {
+    const prevNonce = await sdk.rpc.chain.getNonce(payer);
+    const pendingParcels = await sdk.rpc.chain.getPendingParcels();
+    const payerParcels = pendingParcels.filter((parcel) => parcel.getSignerAddress().value === payer );
+    const nextNonce = await sdk.rpc.chain.getNonce(payer);
+
+    if (nextNonce > prevNonce) {
```
Comment (majecty) [inline index.ts]: What is the purpose of this code?
`prevNonce` and `nextNonce` are acquired by the same method.
When the condition `nextNonce > prevNonce` became true?

**Review by majecty — CHANGES_REQUESTED** [review]

**Comment by majecty** [comment]
Let's have a better commit message.
How about adding "nonce is calculated using pending parcels" information to the message?

File: index.ts (hunk 1 above)
Comment (posgnu) [reply index.ts]: When the block is mined before `sdk.rpc.chain.getPendingParcels()` is called, `nextNonce` became bigger than `prevNonce`

Comment (posgnu) [reply index.ts]: I made it more simple

**Comment by posgnu** [comment]
@majecty great

**Comment by majecty** [comment]
LGTM

**Review by majecty — APPROVED** [review]

**Pull request merged**